iceoryx_examples/icedelivery/README.md

```diff
@@ -242,25 +227,62 @@ The typed publisher application is an example for a high-level user API and does
 described before. In this summary, just the differences to the prior publisher application are described.
 
 Starting again with the includes, there is now a different one:
+
 ```cpp
 #include "iceoryx_posh/popo/publisher.hpp"
 ```
 
 When it comes to the runtime, things are the same as in the untyped publisher. However, a typed publisher object is
-created
+created and the transmitted data type is provided as template parameter:
+
 ```cpp
 iox::popo::Publisher<RadarObject> publisher({"Radar", "FrontLeft", "Object"});
 ```
 
+The topic type must be default- and copy-constructible when the typed API is used.
+
 A similar while-loop is used to send the data to the subscriber. In contrast to the untyped publisher the typed one
-offers two additional possibilities
+offers three additional possibilities.
+
+#### #2 Functional approach with loaning
+
+Usage #2 constructs the data type with the values provided in loan:
+
+```cpp
+auto result = publisher.loan(ct, ct, ct);
+if (!result.has_error())
+{
+    auto& sample = result.value();
+    sample->x = ct;
+    sample->y = ct;
+    sample->z = ct;
+    sample.publish();
+}
+else
+{
+    // ..
+}
+```
+
+One might wonder what the type of the variable `sample` is? It is `iox::popo::Sample<RadarObject>`. This class behaves
+similar to a [`std::unique_ptr`](https://en.cppreference.com/w/cpp/memory/unique_ptr) and makes sure that the ownership
+handling is done automatically and memory is freed when going out of scope on subscriber side.
+
+#### #4 Publish by copy
+
+Usage #4 does a copy-and-publish in one call. This should only be used for small data types, as otherwise copies can
+lead to a larger runtime.
+
 ```cpp
 auto object = RadarObject(ct, ct, ct);
 publisher.publishCopyOf(object).or_else([](iox::popo::AllocationError) {
     // Do something with error.
 });
 ```
-This should only be used for small data types, as otherwise copies can lead to a larger runtime.
+
+#### #5 Publish the result of a computation
+
+Usage #5 can be useful if you have a callable e.g. a function or [functor](https://en.wikipedia.org/wiki/Function_object#In_C_and_C++) should be always called
 
```

Comment:
The required signature of the functor should be stated and explained here! `void(SampleType*)`. Additionally, I would explain roughly how it works:
1. publisher loans sample from shared memory
2. if loaning was successful a pointer to the sample is provided to the functor
3. if loaning was not successful the functor is not called and an error is returned to the user

Reply:
Done